File: server/services/gateway/models/auth_user.go
package models

import "github.com/golang-jwt/jwt/v5"

type AuthUser struct {
	UserID      string `json:"userId"`
	DisplayName string `json:"displayName"`
	Email       string `json:"email"`
	PictureURL  string `json:"pictureUrl"`
}

func ClaimsToAuthUser(claims jwt.MapClaims) AuthUser {
	userID := claims["userId"].(string)
	email := claims["email"].(string)
	displayName := claims["displayName"].(string)
	pictureURL := claims["pictureUrl"].(string)

	return AuthUser{
		UserID:      userID,
		Email:       email,
		DisplayName: displayName,
		PictureURL:  pictureURL,
	}
}

File: apps/mobile/lib/api/models/auth_user.dart
import "package:hive/hive.dart";

part "auth_user.g.dart";

@HiveType(typeId: 16)
class AuthUser {
  @HiveField(0)
  final String userId;
  @HiveField(1)
  final String email;
  @HiveField(2)
  final String displayName;
  @HiveField(3)
  final String? pictureUrl;

  const AuthUser({
    required this.userId,
    required this.email,
    required this.displayName,
    required this.pictureUrl,
  });

  factory AuthUser.fromJson(Map<String, dynamic> json) {
    return AuthUser(
      userId: json["userId"] as String,
      email: json["email"] as String,
      displayName: json["displayName"] as String,
      pictureUrl: json["pictureUrl"] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "userId": userId,
      "email": email,
      "displayName": displayName,
      "pictureUrl": pictureUrl,
    };
  }
}
